### PR TITLE
Minor improvements.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

File: tests/python/test_multidevice.py [hunk 1]
```diff
@@ -34,34 +34,27 @@ def test_pointwise(multidevice_test):
...
             self.sched.parallelize(self.t0, 0, nvfuser.ParallelType.mesh_x)
 
     fn = MultiDeviceModel()
-
-    outputs = fn.execute(inputs)
-
-    for i in range(3):
-        outputs = fn.execute(inputs)
-
-    assert (inputs[0].relu() * 2).allclose(outputs[0][rank])
```
Comment (wujingyue) [inline tests/python/test_multidevice.py:67]: This checks only a slice of the output and therefore doesn't cover allgather.

File: tests/python/test_multidevice.py [hunk 2]
```diff
@@ -34,34 +34,27 @@ def test_pointwise(multidevice_test):
...
             self.sched._set_device_mesh(self.t1, mesh)
             self.sched._set_device_mesh(self.t2, mesh)
             self.sched.parallelize(self.t0, 0, nvfuser.ParallelType.mesh_x)
 
     fn = MultiDeviceModel()
-
-    outputs = fn.execute(inputs)
-
-    for i in range(3):
-        outputs = fn.execute(inputs)
```
Comment (wujingyue) [inline tests/python/test_multidevice.py:65]: I don't see why this is needed.

File: tests/python/test_multidevice.py [hunk 3]
```diff
@@ -34,34 +34,27 @@ def test_pointwise(multidevice_test):
...
             self.t0 = self.define_tensor(
-                (num_devices, 4), (False, False), dtype=DataType.Float
+                (-1, -1), contiguity=(False, False), dtype=DataType.Float
             )
             self.t1 = self.ops.relu(self.t0)
             self.t2 = self.ops.add(self.t1, self.t1)
             self.add_output(self.t2)
 
         def multidevice_schedule(self):
-            mesh = self.sched._create_device_mesh(tuple(range(num_devices)))
```
Comment (wujingyue) [inline tests/python/test_multidevice.py:54]: The code runs fine without `tuple`.

File: tests/python/test_multidevice.py [hunk 4]
```diff
@@ -34,34 +34,27 @@ def test_pointwise(multidevice_test):
...
+    torch.manual_seed(0)
+    unsharded_input = torch.randn(num_devices, 4, device="cuda")
+    sharded_input = unsharded_input[rank : rank + 1]
 
-    # dynamic shape isn't supported;
-    # scalar inputs isn't supported;
     class MultiDeviceModel(FusionDefinition):
         def definition(self):
             self.t0 = self.define_tensor(
-                (num_devices, 4), (False, False), dtype=DataType.Float
```
Comment (wujingyue) [inline tests/python/test_multidevice.py:47]: So we exercise dynamic shapes, which seem to work just fine. I don't remember why we added the comments above.

**Comment by wujingyue** [comment]
!build

File: tests/python/test_multidevice.py [hunk 5]
```diff
@@ -34,34 +34,27 @@ def test_pointwise(multidevice_test):
...
+                (-1, -1), contiguity=(False, False), dtype=DataType.Float
```
Comment (jjsjann123) [inline tests/python/test_multidevice.py:45]: Does this work with `-np 1`?

Not that we have to have that in the test, but I thought the CI script has that... (I could be misreading that)

**Review by jjsjann123 — APPROVED** [review]
this one looks cleaner than mine. Let's merge yours instead.

**Comment by jjsjann123** [comment]
How nice of you, this is merging into the other PR. :bow:

File: tests/python/test_multidevice.py (hunk 5 above)
Comment (jjsjann123) [reply tests/python/test_multidevice.py:45]: NVM, this seems to work fine. :clap:

**Pull request merged**